src/coreclr/gc/gc.cpp

```diff
@@ -48186,6 +48186,42 @@ HRESULT GCHeap::Initialize()
     {
         return E_OUTOFMEMORY;
     }
+
+    /*
+     * Allocation requests less than loh_size_threshold will be allocated on the small object heap.
+     * For regions, we constrainted that objects must stay on a single region, so it is impossible to allocate
+     * an object larger than the available space on a single region.
+     *
+     * Note that that available space on the region is not the region size, but a bit smaller.
+     * The adjustment formula is based on these three implementation details:
+     *
+     * 1.) heap_segment_mem is set to the new pages + sizeof(aligned_plug_and_gap) in make_heap_segment,
+     *     effectively making the bytes before that unusable for objects.
+     * 2.) a_fit_segment_end_p set pad to Align(min_obj_size, align_const), effectively making some end bytes
+     *     unusable
+     * 3.) a_size_fit_p requires the available space to be larger than the allocated size + Align(min_obj_size, align_const)
+     *     again, effectively making some end bytes unusable.
```

Comment:
nit - I think it's fine to delete all the lines that say "effectively making some bytes unusable" since you are talking about how the available size is smaller.